gammapy/maps/hpxnd.py

```diff
@@ -452,6 +457,130 @@ def smooth(self, width, kernel="gauss"):
 
         return self._init_copy(data=smoothed_data)
 
+    def convolve_wcs(self, kernel, use_fft=True, **kwargs):
+        """
+        Convolve map with a WCS kernel.
+
+        It projects the map into a WCS geometry, convolves with a WCS kernel and
+        projects back into the initial Healpix geometry.
+
+        If the kernel is two dimensional, it is applied to all image planes likewise.
+        If the kernel is higher dimensional it must match the map in the number of
+        dimensions and the corresponding kernel is selected for every image plane.
+
+        Parameters
+        ----------
+        kernel : `~gammapy.irf.PSFKernel`
+            Convolution kernel. The pixel size must be upsampled by a factor 2 or bigger
+            with respect to the input map to prevent artifacts in the projection.
+        use_fft : bool
+            Use `scipy.signal.fftconvolve` if True (default)
+            and `ndi.convolve` otherwise.
+        kwargs : dict
+            Keyword arguments passed to `scipy.signal.fftconvolve` or
+            `ndi.convolve`.
+
+        Returns
+        -------
+        map : `HpxNDMap`
+            Convolved map.
+        """
+        if self.geom.width > 10*u.deg:
+            log.warning("Convolution via WCS projection is not recommended for large maps. "\
+                "Perhaps the method `convolve_full()` is more suited for this case.")
+        wcs_geom = kernel.psf_kernel_map.geom.to_image()
+        wcs_size = np.min(wcs_geom._cdelt)
```

Comment:
There is `WcsGeom.pixel_scales` as well. It might be better to be conservative here and choose `np.max()`

Reply:
oops you're right, that is what I intended to do - bad brain. I'll fix it